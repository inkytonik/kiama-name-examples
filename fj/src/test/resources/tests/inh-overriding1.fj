class A extends Object {
  A() {
    super();
  }
  A mk() {
    return new A();
  }
}
class B extends A {
  B() {
    super();
  }
  A mk() {
    return new B();
  }
}
new B().mk()
