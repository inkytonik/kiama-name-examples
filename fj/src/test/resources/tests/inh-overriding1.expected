src/test/resources/tests/inh-overriding1.fj:6:12:info: new A(): A
    return new A();
           ^
src/test/resources/tests/inh-overriding1.fj:14:12:info: new B(): B
    return new B();
           ^
src/test/resources/tests/inh-overriding1.fj:17:1:info: new B().mk ...: A
new B().mk()
^
src/test/resources/tests/inh-overriding1.fj:17:1:info: new B(): B
new B().mk()
^
A
